tests/nn/misc/test_checkpoint_activations.py

```diff
@@ -252,3 +252,28 @@ def train(model, in1, in2):
         if no_cpt[key] != cpt[key]:
             print(no_cpt, cpt)
             assert 0
+
+
+def test_list_input():
+    class Model(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.conv = nn.Linear(2, 2)
+
+        def forward1(self, x):
+            x[0] = self.conv(x[0])
+            return x
+
+        def forward(self, x):
+            y = []
+            y.append(self.conv(x[0]))
+            print("XXX y", y[0].requires_grad, y[0].shape)
+            return y
+
+    model = nn.Sequential(checkpoint_wrapper(Model().cuda()), Model().cuda())
+    in_data = torch.rand(4, 2).requires_grad_(True).cuda()
```

Comment:
requires_grad is not always enough, if the input is int/long autograd can prune the node from the compute graph and a backward hook would be skipped. Not saying that it's exactly what happens here, just a knee jerk reaction

Reply:
Yeah. Doesn’t seem to be the case here though

Reply:
Can you move the input to cuda() and then set the requires_grad field? cuda() should give you a copy of the tensor on the GPU but I've run into instances with Offload where the property of activation tensors was erased when I moved it from CPU->GPU.

Reply:
I have tested this. It seems to make no difference. I think requires_grad is kept when you move the tensor but I do think some other properties are not kept when you move the tensor.